File: src/ts/3._stack.ts
// Estructura de datos
/** Stack */

const stack: string[] = [];

// push: apilamos tres acciones
stack.push('Escribir Hola');
stack.push('Escribir mundo');
stack.push('Colocar negrita');
stack.push('Escribir Mi primer');
stack.push('Escribir parrafo');

// pop: sacamos el stack de arriba (la ultima que entra, la primera que sale)
let action = stack.pop();
console.log(action);

// siguiente stack
action = stack.pop();
console.log(action);

File: src/python/3._stack.py
# Estructura de datos
# Stack (Pila)
# LIFO (Last In First Out) - El último en entrar es el primero en salir

pila: list[str] = []

# apilamos elementos
pila.append("Escribir un libro")
pila.append("Escribir el encabezado")
pila.append("Escribir Hola")
pila.append("Escribir este es")
pila.append("Escribir mi primer libro")

# sacamos elementos de arriba
accion: str = pila.pop()
print(accion)  # Escribir mi primer libro

# siguiente elemento
accion = pila.pop()
print(accion)  # Escribir este es
